<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:dialogs="clr-namespace:fin.ui.avalonia.dialogs"
        xmlns:buttons="clr-namespace:fin.ui.avalonia.buttons"
        xmlns:controls="clr-namespace:fin.ui.avalonia.controls"
        mc:Ignorable="d"
        Title="Exception"
        MinWidth="300"
        MinHeight="100"
        Width="600"
        CanResize="True"
        SizeToContent="Height"
        x:Class="fin.ui.avalonia.dialogs.ExceptionDialog"
        x:DataType="dialogs:ExceptionDialogViewModel">
  <Design.DataContext>
    <dialogs:ExceptionDialogViewModelForDesigner />
  </Design.DataContext>

  <Border Padding="8">
    <StackPanel>
      <TextBlock>Oops, an exception occurred:</TextBlock>
      <controls:CodeViewer Source="{Binding Path=Exception}" />
      <buttons:ReportIssueButton 
          DataContext="{Binding ReportIssueButton}" 
          Margin="0 6 0 0"
          HorizontalAlignment="Center" />
    </StackPanel>
  </Border>
</Window>